### PR TITLE
ui: Show user friendly time labels

### PR DESCRIPTION
**Pull request by innermous**
Title stands for itself.

### CODE REVIEW

File: src/ui/settingsdialog.cpp [hunk 1]
```diff
@@ -68,6 +68,24 @@ const char DocsetNameProperty[] = "docsetName";
 const char DownloadTypeProperty[] = "downloadType";
 const char DownloadPreviousReceived[] = "downloadPreviousReceived";
 const char ListItemIndexProperty[] = "listItem";
+
+QString timeSinceLastUpdate(const QDateTime &updateTime)
+{
+    int diff = updateTime.secsTo(QDateTime::currentDateTime());
+
+    if (diff <= 5)      return QStringLiteral("just now");
+    if (diff <= 20)     return QStringLiteral("%1 seconds ago").arg(diff);
```
Comment (trollixx) [inline src/ui/settingsdialog.cpp]: Please make strings [translation friendly](http://doc.qt.io/qt-5/i18n-source-translation.html#handling-plurals).

File: src/ui/settingsdialog.cpp [hunk 2]
```diff
@@ -68,6 +68,24 @@ const char DocsetNameProperty[] = "docsetName";
 const char DownloadTypeProperty[] = "downloadType";
 const char DownloadPreviousReceived[] = "downloadPreviousReceived";
 const char ListItemIndexProperty[] = "listItem";
+
+QString timeSinceLastUpdate(const QDateTime &updateTime)
+{
+    int diff = updateTime.secsTo(QDateTime::currentDateTime());
```
Comment (trollixx) [inline src/ui/settingsdialog.cpp]: Could be `const`. Sorry, nitpicking.

**Comment by innermous** [comment]
Will send changes in afternoon.

File: src/ui/settingsdialog.cpp (hunk 2 above)
Comment (innermous) [inline src/ui/settingsdialog.cpp]: Oh yeah, const... sometimes mistakes are pretty stupid when it's 6am :santa:

**Comment by trollixx** [comment]
I'd be happy to merge this once earlier comments are addressed.

File: src/ui/settingsdialog.cpp [hunk 3]
```diff
@@ -68,6 +68,24 @@ const char DocsetNameProperty[] = "docsetName";
...
+
+    if (diff <= 5)      return QStringLiteral("just now");
```
Comment (trollixx) [inline src/ui/settingsdialog.cpp]: This should be `QObject::tr("just now")`

File: src/ui/settingsdialog.cpp (hunk 1 above)
Comment (trollixx) [inline src/ui/settingsdialog.cpp]: `QObject::tr("%n second(s) ago", "", diff)`

Comment (trollixx) [inline src/ui/settingsdialog.cpp]: The [article](http://doc.qt.io/archives/qq/qq19-plurals.html) linked in the above documentation page has more information.

File: src/ui/settingsdialog.cpp [hunk 4]
```diff
@@ -68,6 +68,24 @@ const char DocsetNameProperty[] = "docsetName";
...
+    if (diff <= 20)     return QStringLiteral("%1 seconds ago").arg(diff);
+    if (diff < 60)      return QStringLiteral("less than a minute ago");
+    if (diff <= 90)     return QStringLiteral("one minute ago");
+    if (diff <= 3540)   return QStringLiteral("%1 minutes ago").arg(std::round(diff / 60));
+    if (diff <= 5400)   return QStringLiteral("1 hour ago");
```
Comment (trollixx) [inline src/ui/settingsdialog.cpp]: Why `one minute ago`, but `1 hour ago`? I think texts should be consistent between units.

**Comment by jalcine** [comment]
Hey @innermous! I'd like to help out nudging this PR along. Anything else outside of the changes @trollixx mentioned blocking you so far?

**Comment by innermous** [comment]
@jalcine go for it :)

**Comment by jalcine** [comment]
Looks like it has a conflict. I can clone and add that patch if need be, @innermous from your fork.

**Comment by trollixx** [comment]
This is an old and conflicting patch, if anyone is interested, please submit a new one.

**Pull request closed without merging**